src/opensourceReminder.job.ts

```diff
@@ -0,0 +1,58 @@
+import config from './config';
+import { scheduleJob } from './scheduler.service';
+import { getConfiguredGuild } from './commandHandlers/guild.service';
+import { db } from './firebase';
+import { UserSubscriptions } from './config';
+import { log } from './logger';
+
+export function scheduleOpenSourceReminder() {
+    scheduleJob(config.OPENSOURCE_JOB_CRON_TIME, reminderCallback, 'opensource-reminder');
+}
+
+/**
+ * Send a DM to users that are subscribed to this reminder with a message.
+ * The user's subscriptions are kept on the "users_subscriptions" DB collection.
+ */
+async function reminderCallback() {
+    // TODO: use the GitHub API to check if the user has already contributed on this day.
+    // TODO: if he has, then send a rewarding positive message instead of a reminder.
+
+    // Get the list of users that are subscribed to the open source subscription
+    const querySnapshot = await db.collection('users_subscriptions').get();
+    const userNamesToSendMessage = querySnapshot
+        .docs
+        .filter(doc => {
+            const userSubscription = doc.data() as UserSubscriptionDb;
+            return userSubscription.subscriptions.includes(UserSubscriptions.OPEN_SOURCE);
+        })
+        .map(doc => (doc.data() as UserSubscriptionDb).username);
+    const guild = getConfiguredGuild();
+    const allMembers = await guild!.members.fetch();
+    const subscribedUsers = allMembers.filter(member => userNamesToSendMessage.includes(member.user.username));
+
+    // Send the messages in parallel to subscribed users
+    log.info(`Starting to send open source reminder messages to ${subscribedUsers.size} subscribed users`);
+    const promises = subscribedUsers.map(member => {
+        const userId = member.user.id;
+        try {
+            log.info(`Sending a open source reminder message to user ${userId}`, userId);
+            member.user.send('Hey :wave:, this is a friendly reminder for you to contribute to open source today. Let\'s get those green squares together :smiley:!');
+            return;
+        } catch (e) {
+            log.error(`ERROR: Sending the open source reminder message to the user with id: ${userId}`, e.message);
+            log.error(e);
+
+            // Note: we log that an error happened when sending a DM to the user, but we don't propagate the error
+            // so that Promise.all doesn't get rejected.
+            return;
```

Comment:
would a `Promise.resolve(1);` be tidier here?

Reply:
Hmm, in all fairness, with using a .map() you should really not end up in this catch statement, `return true;` or similar may suffice here,.

Reply:
Hi @stemount, thanks for the comment and sorry for the late response. Yes, I agree with you this code could be tidier, so we can separate this into the problem and solution.
Regarding the fact that the code won't end up in the catch statement, I think it can if a network error occurs, or any other exception in the library code. If you check [these docs](https://discord.js.org/#/docs/main/stable/class/GuildMember?scrollTo=send) you'll see the examples adding a catch clause, and I assume some users might be able to disable bots to send messages like this.

**Current Problem**
If we currently hover over the promises variable, this is its type: `void[]` . So it's not clear that the values of the array are in fact promises. _Note:_ if I understood your comment correctly, this is the problem_

**Proposed Solutions**
If we change to `return true` it has this type:
![imagem](https://user-images.githubusercontent.com/18630253/90317480-c9285580-df21-11ea-972f-c1602f06f09b.png)

If we change to your first suggestion it has this type:
![imagem](https://user-images.githubusercontent.com/18630253/90317523-1278a500-df22-11ea-9fb9-f87a87d4c630.png)

or in case we change both return statements, this type:
![imagem](https://user-images.githubusercontent.com/18630253/90317814-e2320600-df23-11ea-8ed3-34cda8e40e2e.png)

I still think we could improve it to be `Promise<void>[]`, so what do you think of addind the `async` keyword, basically saying a promise is returned, not that we'll use the `await` keyword. Also, imo we should use `void` instead of returning a number because the value of the promise doesn't matter. I think with `void` it becomes explicit that these values won't be used.
![imagem](https://user-images.githubusercontent.com/18630253/90317573-6d120100-df22-11ea-8d4e-933c34cb8210.png)

What solution do you think we should adopt @stemount ?

Reply:
I was wrong when I said the code on the catch clause does run. It doesn't because I don't await the promise returned by `member.user.send` 😅, so I'll be fixing that 👍 

For reference here is the related comment: https://github.com/EddieJaoudeCommunity/EddieBot/pull/154#discussion_r473401130